common/flatpak-run-wayland.c

```diff
@@ -281,6 +299,11 @@ flatpak_run_add_wayland_args (FlatpakBwrap *bwrap,
       flatpak_bwrap_add_runtime_dir_member (bwrap, wayland_display);
     }
 
+#ifdef ENABLE_WAYLAND_SECURITY_CONTEXT
+  if (security_context_available)
+    return TRUE;
+#endif
```

Comment:
I'm unsure whether we should be bailing out early in the `security_context_available = TRUE` case, or continuing into the rest this function, which is code added by 0402e161 (#5615) to deal with inheritance of `$WAYLAND_SOCKET`. Having the early-return here preserves current functionality, so that's what I've gone with.

If a Wayland expert (@emersion? @swick?) thinks it would be better to drop this early-return and continue into `$WAYLAND_SOCKET` then I think that should be a third commit following the two I already have here.

Reply:
The security context and WAYLAND_SOCKET are mutually exclusive features. The only way to preserve WAYLAND_SOCKET is by not applying the security context. As a caller of flatpak-run its your own responsibility to make sure that the connections to WAYLAND_SOCKET don't expose anything that app shouldn't have access to.

All of that makes me think that we should indeed drop this early returns.

Reply:
> The security context and WAYLAND_SOCKET are mutually exclusive features. The only way to preserve WAYLAND_SOCKET is by not applying the security context.

Yes, and conversely, the only way to enforce use of the security-context-v1 per-app socket is by not having a WAYLAND_SOCKET.

> All of that makes me think that we should indeed drop this early returns.

Done in 0e55b1a (currently untested, I'll try to set up a compatible compositor so I can try this). This gives us a way to tie-break which factor the app maintainer or user considers to be more important: "programs launched with a special WAYLAND_SOCKET should use it" (`--socket=inherit-wayland-socket`) or "security-context-v1 should be used if available" (`--nosocket=inherit-wayland-socket`, default).